### PR TITLE
Create temp training table and run PAI tensorflow step

### LINKED ISSUE #1335: Create a temp MaxCompute table from SELECT statement and run the training
Part of https://github.com/sql-machine-learning/sqlflow/issues/1321

For submitting training jobs to PAI and read data from MaxCompute, we need to generate two Couler steps from one train SQL:

1. run `create table tmp_training_table from (select ...)` to create a temp table.
2. use the temp table in `TableRecordDataset` when submitting a PAI training job.

### PR DESCRIPTION
**Pull request by typhoonzero**
Resolve https://github.com/sql-machine-learning/sqlflow/issues/1335
part of https://github.com/sql-machine-learning/sqlflow/issues/1321

Should depend on https://github.com/sql-machine-learning/sqlflow/pull/1384

### CODE REVIEW

File: scripts/test_workflow.sh [hunk 1]
```diff
@@ -69,7 +69,7 @@ check_ret $? "Test Couler failed"
 ############# Run SQLFLow test with Argo Mode #############
 function test_workflow() {
     # start a SQLFlow MySQL Pod with testdata
-    kubectl run mysql --port 3306 --env="SQLFLOW_MYSQL_HOST=0.0.0.0" --env="SQLFLOW_MYSQL_PORT=3306" --image=sqlflow/sqlflow --command -- bash /start.sh mysql
+    kubectl run mysql --port 3306 --env="SQLFLOW_MYSQL_HOST=0.0.0.0" --env="SQLFLOW_MYSQL_PORT=3306" --image=sqlflow:submitter --command -- bash /start.sh mysql
```
Comment (Yancey0623) [inline scripts/test_workflow.sh]: Maybe can use the env `SQLFLOW_WORKFLOW_STEP_IMAGE` here which been configurated in 
https://github.com/sql-machine-learning/sqlflow/blob/9d64323bef36fc3bc87c5f51f94917f5c2322ce0/.travis.yml#L67

File: scripts/test_workflow.sh [hunk 2]
```diff
@@ -94,4 +94,10 @@ function test_workflow() {
 test_workflow
 check_ret $? "Test SQLFLow workflow failed"
 
+# test submit pai job using argo workflow mode
+if [ $SQLFLOW_submitter == "pai" ]; then
```
Comment (Yancey0623) [inline scripts/test_workflow.sh:98]: This is not work in the current PR? Maybe we need to add `SQLFLOW_submitter=pai` in the `travis.yaml`

Comment (typhoonzero) [reply scripts/test_workflow.sh:98]: `SQLFLOW_submitter=pai` tests now can only run in the internal environment

File: scripts/test_workflow.sh [hunk 3]
```diff
@@ -69,7 +69,7 @@ check_ret $? "Test Couler failed"
 ############# Run SQLFLow test with Argo Mode #############
 function test_workflow() {
     # start a SQLFlow MySQL Pod with testdata
-    kubectl run mysql --port 3306 --env="SQLFLOW_MYSQL_HOST=0.0.0.0" --env="SQLFLOW_MYSQL_PORT=3306" --image=sqlflow/sqlflow --command -- bash /start.sh mysql
+    kubectl run mysql --port 3306 --env="SQLFLOW_MYSQL_HOST=0.0.0.0" --env="SQLFLOW_MYSQL_PORT=3306" --image=${SQLFLOW_WORKFLOW_STEP_IMAGE} --command -- bash /start.sh mysql
```
Comment (Yancey0623) [inline scripts/test_workflow.sh:72]: Maybe we can set the default docker image by: `${SQLFLOW_WORKFLOW_STEP_IMAGE:-sqlflow/sqlflow}` .

**Review by Yancey0623 — APPROVED** [review]
LGTM with tiny comment.

**Pull request merged**